root/md.html

```diff
@@ -0,0 +1,51 @@
+<%- rss = 'distribution/' _ file.distribution %>
+<%- title = file.path %>
+<div class="breadcrumbs">
+    <a data-keyboard-shortcut="g s" href="/source/<% base = [file.author, file.release].join("/"); base %>"><% [file.author, file.release].join(" / ") %></a>
+    <% doc_view_url = [base,file.path].join("/") %>
+    <% raw_url = [api_external_secure,'source',base,file.path].join("/") %>
+  <% FOREACH part IN file.path.split("/"); base = base _ "/" _ part -%>
+    / <% UNLESS loop.last %><a href="/source/<% base %>"><% part %></a><% ELSE %><% part %><% END %>
+    <% END %>
+</div>
+
+<ul class="nav-list slidepanel">
+  <li class="visible-xs">
+    <% INCLUDE mobile/toolbar-search-form.html %>
+  </li>
+  <li class="nav-header">Tools</li>
+  <li>
+    <a data-keyboard-shortcut="g d" href="/release/<% file.author %>/<% file.release %>"><i class="fa fa-info-circle fa-fw black"></i>Release Info</a>
+  </li>
+  <% IF file.documentation %>
+  <li>
+    <a href="/pod/<% file.documentation %>"><i class="fa fa-book fa-fw black"></i>Module Documentation</a>
+  </li>
+  <% ELSIF file.slop %>
+  <li>
+    <a href="/pod/release/<% doc_view_url %>"><i class="fa fa-book fa-fw black"></i>Documentation View</a>
+  </li>
+  <% END %>
+  <li>
+    <a data-keyboard-shortcut="g a" href="/author/<% file.author %>"><i class="fa fa-user fa-fw black"></i>Author</a>
+  </li>
+  <li>&nbsp;</li>
+  <li><a href="<% raw_url %>"><i class="fa fa-file-text-o fa-fw black"></i>Raw code</a></li>
+  <li>
+    <a href="/raw/<% file.author %>/<% file.release %>/<% file.path %>?download=1"><i class="fa fa-download fa-fw black"></i>Download</a>
+  </li>
+  <% IF file.sloc > 0 %>
+  <li><button class="btn-link pod-toggle<% IF file.sloc > 0; " pod-hidden"; END %>" onclick="togglePod()"><i class="fa fa-exchange fa-fw black"></i><span class="hide-pod">Hide</span><span class="show-pod">Show</span> Pod</button></li>
+  <% END %>
+  <li class="nav-header">Info</li>
+  <li><% file.sloc %> lines of code</li>
+  <li><% file.slop %> lines of pod</li>
+  <li><% file.stat.size | format_bytes %> bytes</li>
+</ul>
+
+<div class="content">
+<% USE MultiMarkdown(heading_ids => 1) -%>
+<% FILTER multimarkdown %>
+<% source %>
+<% END %>
+</div>
```

Comment:
This template seems to resemble `root/source.html` very closely.  I wonder if we could share the same template and just add the filter conditionally?

Reply:
Agreed. Taking it one step farther, do you think that this functionality belongs within the Source controller, just as a special case? Or is the new Md controller appropriate? I've gone back and forth several times now.

Reply:
I think the Source controller is probably a good place for this.  I think it could go either way, but putting it in the Source controller likely means less lines of code.